graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetsResource.java

```diff
@@ -211,17 +211,20 @@ public IndexSetSummary update(@ApiParam(name = "id", required = true)
                                   @ApiParam(name = "Index set configuration", required = true)
                                   @Valid @NotNull IndexSetUpdateRequest updateRequest) {
         checkPermission(RestPermissions.INDEXSETS_EDIT, id);
-        if (!id.equals(updateRequest.id())) {
-            throw new ClientErrorException("Mismatch of IDs in URI path and payload", Response.Status.CONFLICT);
-        }
 
         final IndexSetConfig oldConfig = indexSetService.get(id)
                 .orElseThrow(() -> new NotFoundException("Index set <" + id + "> not found"));
 
-        final IndexSetConfig savedObject = indexSetService.save(updateRequest.toIndexSetConfig(oldConfig));
         final IndexSetConfig defaultIndexSet = indexSetService.getDefault();
+        final boolean isDefaultSet = oldConfig.equals(defaultIndexSet);
+
+        if (isDefaultSet && !updateRequest.isWritable()) {
```

Comment:
This writable check here isn't enough to prevent readonly indices to become default, actually switching the current default index set is done in `setDefault` below, which is being used by the UI.